cluster-autoscaler/core/scaledown/budgets/budgets_test.go

```diff
@@ -298,6 +362,62 @@ func TestCropNodesToBudgets(t *testing.T) {
 			wantEmpty:                generateNodeGroupViewList(testNg, 0, 4),
 			wantDrain:                generateNodeGroupViewList(testNg, 0, 2),
 		},
+		"empty&drain atomic nodes with deletions in progress, overall budget exceeded, both empty&drain nodes fit": {
+			emptyDeletionsInProgress: 5,
+			drainDeletionsInProgress: 2,
+			empty:                    generateNodeGroupViewList(atomic4, 0, 4),
+			drain:                    generateNodeGroupViewList(atomic3, 0, 3),
+			wantEmpty:                generateNodeGroupViewList(atomic4, 0, 4),
+			wantDrain:                []*NodeGroupView{},
+		},
+		"empty&drain atomic nodes in same group with deletions in progress, both empty&drain nodes fit": {
+			emptyDeletionsInProgress: 5,
+			drainDeletionsInProgress: 2,
+			empty:                    generateNodeGroupViewList(atomic3, 0, 2),
+			drain:                    generateNodeGroupViewList(atomic3, 2, 3),
+			wantEmpty:                generateNodeGroupViewList(atomic3, 0, 2),
+			wantDrain:                generateNodeGroupViewList(atomic3, 2, 3),
+		},
+		"empty&drain atomic nodes in same group with deletions in progress, overall budget exceeded, both empty&drain nodes fit": {
+			emptyDeletionsInProgress: 5,
+			drainDeletionsInProgress: 2,
+			empty:                    generateNodeGroupViewList(atomic8, 0, 6),
+			drain:                    generateNodeGroupViewList(atomic8, 6, 8),
+			wantEmpty:                generateNodeGroupViewList(atomic8, 0, 6),
+			wantDrain:                generateNodeGroupViewList(atomic8, 6, 8),
+		},
+		"empty&drain atomic nodes in same group with deletions in progress, drain budget exceeded, both empty&drain nodes fit": {
+			emptyDeletionsInProgress: 2,
+			drainDeletionsInProgress: 2,
+			empty:                    generateNodeGroupViewList(atomic5, 0, 1),
+			drain:                    generateNodeGroupViewList(atomic5, 1, 5),
+			wantEmpty:                generateNodeGroupViewList(atomic5, 0, 1),
+			wantDrain:                generateNodeGroupViewList(atomic5, 1, 5),
+		},
+		"empty&drain regular and atomic nodes with deletions in progress, overall budget exceeded, only atomic is drained": {
```

Comment:
>only atomic is drained

nit: it seems that in this test case nothing is drained, only the entirely empty atomic group is removed?

Reply:
Fixed the test case name